### PR TITLE
Remove references to s1053, use installdir instead

### PR DESCRIPTION
**Pull request by mcgibbon**
## Purpose

This PR restores the "installdir" environment variable to machineEnvironment.sh in buildenv, and makes more consistent use of it inside our tests which make use of buildenv.

Depends on buildenv PR https://github.com/ai2cm/buildenv/pull/33, https://github.com/ai2cm/buildenv/pull/34.

## Infrastructure changes:

- virtualenv scripts use the ${installdir} environment variable instead of a hard-coded path to `/project/s1053/install` on daint
- requirements_daint.txt is removed, mpi4py is instead installed in daint_venv
- unused references to TEST_DATA_DIR inside `/project/s1053` are removed

## Checklist
Before submitting this PR, please make sure:

- [ ] You have followed the coding standards guidelines established at [Code Review Checklist](https://paper.dropbox.com/doc/Code-Review-Checklist--BD7zigBMAhMZAPkeNENeuU2UAg-IlsYffZgTwyKEylty7NhY).
- [ ] Docstrings and type hints are added to new and updated routines, as appropriate
- [ ] All relevant documentation has been updated or added (e.g. README, CONTRIBUTING docs)
- [ ] Unit tests are added or updated for non-stencil code changes

Additionally, if this PR contains code authored by new contributors:

- [ ] The names of all the new contributors have been added to CONTRIBUTORS.md

### CODE REVIEW

**Review by rheacangeo — APPROVED** [review]
awesome, thanks for consolidating all those variables

File: .jenkins/install_virtualenv.sh [hunk 1]
```diff
@@ -11,7 +11,7 @@ exitError()
 
 # check a virtualenv path has been provided
 test -n "$1" || exitError 1001 ${virtualenv_path} "must pass an argument"
-wheel_dir=/project/s1053/install/wheeldir
+wheel_dir=${installdir}/wheeldir
```
Comment (rheacangeo) [inline .jenkins/install_virtualenv.sh:14]: this might not be set if machineEnvironment is not run?

Comment (rheacangeo) [reply .jenkins/install_virtualenv.sh:14]: I think you might need a . ${BUILDENV_DIR}/machineEnvironment.sh in this script

**Comment by mcgibbon** [comment]
Launched performance plan https://jenkins.ginko.ch/job/pace-fv3core-performance-test-PR/19/

**Comment by mcgibbon** [comment]
Performance plan passed. Merging.

**Pull request merged**